internal/crypto/tls.go

```diff
@@ -7,30 +7,45 @@ import (
 	oscrypto "github.com/openshift/library-go/pkg/crypto"
 )
 
-func TLSConfigForServer(caConfig, serverConfig *TLSCertificateConfig) (*tls.Config, error) {
+func TLSConfigForServer(caConfig, serverConfig *TLSCertificateConfig) (*tls.Config, *tls.Config, *tls.Config, error) {
 	certBytes, err := oscrypto.EncodeCertificates(serverConfig.Certs...)
 	if err != nil {
-		return nil, err
+		return nil, nil, nil, err
 	}
 	keyBytes, err := PEMEncodeKey(serverConfig.Key)
 	if err != nil {
-		return nil, err
+		return nil, nil, nil, err
 	}
 	cert, err := tls.X509KeyPair(certBytes, keyBytes)
 	if err != nil {
-		return nil, err
+		return nil, nil, nil, err
 	}
 
 	caPool := x509.NewCertPool()
 	for _, caCert := range caConfig.Certs {
 		caPool.AddCert(caCert)
 	}
-	return &tls.Config{
+
+	tlsConfig := &tls.Config{
+		Certificates: []tls.Certificate{cert},
+		MinVersion:   tls.VersionTLS13,
+	}
+
+	agentTlsConfig := &tls.Config{
 		Certificates: []tls.Certificate{cert},
 		ClientCAs:    caPool,
 		ClientAuth:   tls.RequireAndVerifyClientCert,
 		MinVersion:   tls.VersionTLS13,
-	}, nil
+	}
+
+	grpcTlsConfig := &tls.Config{
+		Certificates: []tls.Certificate{cert},
+		ClientCAs:    caPool,
+		ClientAuth:   tls.VerifyClientCertIfGiven,
```

Comment:
another attempt to make grpc endpoit work for both users & devices ... until we have a proper user-facing endpoint.

The mTLS cert is optional. Any connection has to present either valid Auth header (users) or mTLS cert (devices).

@mangelajo WDYT ?